modules/RTC/RTC.js

```diff
@@ -143,6 +220,150 @@ export default class RTC extends Listenable {
     }
 
     /**
+     * Gets streams from specified device types and creates {@code
+     * JitsiLocalTrack} instances for each. This call intentionally ignores
+     * errors for upstream to catch and handle instead.
+     *
+     * @param {Object} options - A hash describing what devices to get and
+     * relevant constraints.
+     * @param {string[]} options.devices - The types of media to capture. Valid
+     * values are "desktop", "audio", and "video".
+     * @returns {Promise} The promise, when successful, will return an array of
+     * JitsiLocalTrack instances. If an error occurs, it will be deferred to the
+     * caller for handling.
+     */
+    static _newObtainAudioAndVideoPermissions(options = {}) {
+        logger.info('Using the new gUM flow');
+
+        const localJitsiTracks = [];
+
+        // Declare private functions to be used in the promise chain below.
+        // These functions are declared in the scope of this function because
+        // they are not being used anywhere else, so only this function needs to
+        // know about them.
+
+        /**
+         * Executes a request for desktop media if specified in options.
+         *
+         * @returns {Promise}
+         */
+        function maybeRequestDesktopDevice() {
+            const devices = options.devices || [ 'audio', 'video' ];
+            const requestedDesktopDevices = devices.filter(
+                device => device === 'desktop');
```

Comment:
since we only filter for a single type, i wonder if it might be clearer to do:

```
const isDesktopDeviceRequsted = devices.indexOf('desktop') !== -1;

return isDesktopDeviceRequested ? ... : ...
```